Update switchbot docs (#38587)
home-assistant/home-assistant.github.io: source/_integrations/switchbot.markdown

  description: Locks' encryption key
{% endconfiguration_basic %}

For instructions on how to obtain the encryption key, see README in [PySwitchbot](https://github.com/Danielhiversen/pySwitchbot#obtaining-locks-encryption-key) project.

> 
> 
> 
> {% include integrations/config_flow.md %}
> 
> 
> 

Reviewer: ```suggestion

{% include integrations/config_flow.md %}

```

> ## Supported Devices
> ### Plugs and switches

Reviewer: ```suggestion
## Supported devices

### Plugs and switches

```

- [Bot (WoHand)](https://switch-bot.com/pages/switchbot-bot)
- [Plug Mini (WoPlug)](https://www.switch-bot.com/products/switchbot-plug-mini)
- [Plug Mini (HomeKit Enabled)](https://www.switch-bot.com/products/switchbot-plug-mini-homekit-enabled)
- [Relay Switch 1](https://www.switch-bot.com/products/switchbot-relay-switch-1)
- [Relay Switch 1PM](https://www.switch-bot.com/products/switchbot-relay-switch-1pm)

> ### Cover

Reviewer: ```suggestion
### Cover

```

- [Curtain (WoCurtain)](https://switch-bot.com/pages/switchbot-curtain) (version 1 & 2)
- [Curtain 3 (WoCtn3)](https://switch-bot.com/pages/switchbot-curtain-3)
- [Blind Tilt (WoBlindTilt)](https://switch-bot.com/pages/switchbot-blind-tilt)
- [Roller Shade](https://www.switch-bot.com/products/switchbot-roller-shade)

> ### Lights

Reviewer: ```suggestion
### Lights

```

- [Color Bulb (WoBulb)](https://switch-bot.com/pages/switchbot-color-bulb)
- [Light Strip (WoStrip)](https://www.switchbot.jp/products/switchbot-strip-light)

> ### Locks

Reviewer: ```suggestion
### Locks

```

- [Lock (WoLock)](https://switch-bot.com/pages/switchbot-lock)
- [Lock Pro (WoLockPro)](https://www.switch-bot.com/pages/switchbot-lock-pro)

> ### Humidifiers

Reviewer: ```suggestion
### Humidifiers

```

- [Humidifier (WoHumi)](https://www.switchbot.jp/products/switchbot-smart-humidifier)

> ### Sensors

Reviewer: ```suggestion
### Sensors

```

- [Meter](https://switch-bot.com/pages/switchbot-meter)
- [Meter Plus (WoSensorTH)](https://switch-bot.com/pages/switchbot-meter-plus)
- [Indoor/Outdoor Meter (WoIOSensorTH)](https://switch-bot.com/pages/switchbot-indoor-outdoor-thermo-hygrometer) 
- [Meter Pro](https://www.switch-bot.com/products/switchbot-meter-pro)
- [Meter Pro CO2 Monitor](https://www.switch-bot.com/products/switchbot-meter-pro-co2-monitor)
- [Contact Sensor (WoContact)](https://switch-bot.com/pages/switchbot-contact-sensor)
- [Motion Sensor (WoPresence)](https://switch-bot.com/pages/switchbot-motion-sensor)
- [Water Leak Detector](https://www.switch-bot.com/products/switchbot-water-leak-detector)
- [Remote (WoRemote)](https://www.switch-bot.com/products/switchbot-remote) (currently only supports battery level monitoring)

> ### Hubs

Reviewer: ```suggestion
### Hubs

```

- [Hub 2 (WoHub2)](https://switch-bot.com/pages/switchbot-hub-2) (currently only supports retrieving sensor data, does not yet support device control)
- [Hub Mini Matter Enabled](https://www.switch-bot.com/products/switchbot-hub-mini-matter-enabled)(currently only supports retrieving sensor data, does not yet support device control)

> ## Supported Functionality

Reviewer: ```suggestion
## Supported functionality

```

### Common

#### Options
- `Retry count`: How many times to retry sending commands to your SwitchBot devices.

#### Attributes
- `last_run_success`: Returns `true` if the last action sent to the SwitchBot succeeded. This attribute is useful for error trapping when Bluetooth connectivity is intermittent. If `false`, see home-assistant.log for specific error messages.

> ### Plugs and switches

Reviewer: ```suggestion
### Plugs and switches

```

Switch entities are added for Bot, Plug Mini, and Relay Switch.

> #### Bot

Reviewer: ```suggestion
#### Bot

```

Password protection: You can set a device password in the SwitchBot app to prevent people nearby take control of your device. When a password is set, you need to enter the correct password in order to add it to the integrations.

Features:
- turn on or off
- press
- get battery level

Attributes:
- `Switch mode`: Specifies the mode of the device. If `true`, the device is in Pull/Retract mode for toggle switches otherwise the device is in momentary switch mode.

> #### Plug Mini

Reviewer: ```suggestion
#### Plug Mini

```

Features:
- turn on or off
- get power consumption readings

> #### Relay Switch 1

Reviewer: ```suggestion
#### Relay Switch 1

```

This is an encrypted device.

Features:
- turn on or off

> #### Relay Switch 1PM

Reviewer: ```suggestion
#### Relay Switch 1PM

```

This is an encrypted device.

Features:
- turn on or off
- get power
- get voltage
- get current

> ### Cover

Reviewer: ```suggestion
### Cover

```

Cover entities are added for Curtain, Curtain 3, Blind Tilt, and Roller Shade.

> #### Curtain

Reviewer: ```suggestion

#### Curtain

```

Features:
- open/close/pause
- set position
- get position
- get light level
- get battery level
- get calibration state

> #### Curtain 3

Reviewer: ```suggestion
#### Curtain 3

```

Features:
- open/close/pause
- set position
- get position
- get light level
- get battery level
- get calibration state

> #### Blind Tilt

Reviewer: ```suggestion
#### Blind Tilt

```

{% endraw %}

#### Roller Shade
The Roller Shade is exposed as a cover entity with control of the position only:

| position | Roller Shade state |
| -------- | ------------------ |
| <=20%    | Open               |
| >20%     | Close              |

Features:
- open/close/pause
- set position
- get position
- get battery level

### Sensors

Sensor entiteis are added for thermometer and hygrometer devices, motion sensor, contact sensor, leak sensor, and remote button.

> #### Meter

Reviewer: ```suggestion
#### Meter

```

Features:
- get temperature
- get humidity
- get battery level

> #### Meter Plus

Reviewer: ```suggestion

#### Meter Plus

```

Features:
- get temperature
- get humidity
- get battery level

> #### Indoor/Outdoor Meter

Reviewer: ```suggestion

#### Indoor/Outdoor Meter

```

Features:
- get temperature
- get humidity
- get battery level

> #### Meter Pro

Reviewer: ```suggestion

#### Meter Pro

```

Features:
- get temperature
- get humidity
- get battery level

> #### Meter Pro CO2 Monitor

Reviewer: ```suggestion

#### Meter Pro CO2 Monitor

```

Features:
- get temperature
- get humidity
- get carbon dioxide
- get battery level

> #### Contact Sensor

Reviewer: ```suggestion

#### Contact Sensor
```

Features:

- open or closed state
- motion detection state
- get battery level

> #### Motion Sensor

Reviewer: ```suggestion

#### Motion Sensor
```

Features:

- motion detection state
- get battery level
#### Water Leak Detector
This is an encrypted device.

Features:
- leak or dry state
- get battery level

> #### Remote

Reviewer: ```suggestion

#### Remote

```

Features:
- get battery level

> ### Lights

Reviewer: ```suggestion
### Lights

```

Light entities are added for Color Bulb and LED Strip Light.

> #### Color Bulb

Reviewer: ```suggestion
#### Color Bulb

```

Features:
- turn on or off
- change brightness
- change color temperature
- change color

> #### LED Strip Light

Reviewer: ```suggestion
#### LED Strip Light

```

Features:
- turn on or off
- change brightness
- change color

### Locks

Note: The integration currently only uses the primary lock state; in dual lock mode, not all things might work properly.

#### Lock

This is an encrypted device.

Features:
- Lock or unlock
- open or closed state
- auto-lock paused state
- calibration state
- get battery level

> #### Lock Pro

Reviewer: ```suggestion

#### Lock Pro
```

This is an encrypted device.

Features:

- Lock or unlock
- open or closed state
- auto-lock paused state
- calibration state
- get battery level

> ### Hubs

Reviewer: ```suggestion

### Hubs
```

#### Remote

Features:
- get battery level

### Lights

Light entities are added for Color Bulb and LED Strip Light.

#### Color Bulb

Features:
- turn on or off
- change brightness
- change color temperature
- change color

#### LED Strip Light

Features:
- turn on or off
- change brightness
- change color

### Locks

Note: The integration currently only uses the primary lock state; in dual lock mode, not all things might work properly.

#### Lock

This is an encrypted device.

Features:
- Lock or unlock
- open or closed state
- auto-lock paused state
- calibration state
- get battery level

#### Lock Pro

This is an encrypted device.

Features:

- Lock or unlock
- open or closed state
- auto-lock paused state
- calibration state
- get battery level

### Hubs

Some of the hubs can be served as a bridge while the sensor data can be retrieved via Bluetooth connection. Hub 2 displays temperature and humidity through a sensor cable. Without a digital display, Hub Mini Matter Enabled can also read from a sensor cable.

#### Hub 2

Features:

- get temperature
- get humidity
- get light level

#### Hub Mini Matter Enabled

Features:

- get temperature
- get humidity

## Data updates

> Devices are polled for data updates every 5 seconds. When you make changes through Home Assistant (e.g., switching a device on), the device's state is updated immediately rather than waiting for the next poll.

Reviewer: ```suggestion
Devices are {% term polling polled %} for data updates every 5 seconds. When you make changes through Home Assistant (for example by switching a device on), the device's state is updated immediately rather than waiting for the next poll.
```

Make sure your devices are powered on and are in range.
{% enddetails %}

## Examples

### Automation ideas

You can create all sort of automations using sensors as triggers and switches, shades, and lights as actions.

> - Turn on or turn off lights and switches when motion is detected
> - Turn on or turn off Bot to control fan, AC remote, circulator when temperature or humidity is too low
> - Close curtain when temperature is too high
> - Turn off Relay Switch when power consumption is too high
> - Change the color of LED Strip Lights or Color Bulb or turn on the circulator using Bot or Relay Switch if carbon dioxide is too high

Reviewer: ```suggestion
- Turn on or turn off lights and switches when motion is detected.
- Turn on or turn off the Bot to control a fan, AC remote, or circulator when temperature or humidity are too low.
- Close the curtain when the temperature is too high.
- Turn off Relay Switch when power consumption is too high.
- Change the color of LED Strip Lights or Color Bulb or turn on the circulator using Bot or Relay Switch if carbon dioxide is too high.
```
